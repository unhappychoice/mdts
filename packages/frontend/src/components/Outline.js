"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const ChevronLeft_1 = __importDefault(require("@mui/icons-material/ChevronLeft"));
const ChevronRight_1 = __importDefault(require("@mui/icons-material/ChevronRight"));
const material_1 = require("@mui/material");
const useOutline_1 = require("../hooks/apis/useOutline");
const Outline = ({ filePath, onItemClick, isOpen, onToggle }) => {
    const { outline, loading, error } = (0, useOutline_1.useOutline)(filePath);
    if (error)
        return (0, jsx_runtime_1.jsxs)("p", { children: ["Error: ", error] });
    return ((0, jsx_runtime_1.jsxs)(material_1.Box, { sx: {
            width: isOpen ? '300px' : '66px',
            bgcolor: 'background.paper',
            p: isOpen ? 2 : 0.5, // Adjust padding when closed
            borderLeft: '1px solid',
            borderColor: 'divider',
            minHeight: '100%',
            flexShrink: 0,
        }, children: [(0, jsx_runtime_1.jsxs)(material_1.Box, { sx: { display: 'flex', justifyContent: 'flex-start', alignItems: 'center', height: '34px', marginTop: isOpen ? '0' : '16px', marginBottom: 2 }, children: [(0, jsx_runtime_1.jsx)(material_1.IconButton, { onClick: onToggle, size: "small", sx: { marginBottom: 0, marginLeft: isOpen ? '0' : '12px' }, children: isOpen ? (0, jsx_runtime_1.jsx)(ChevronRight_1.default, {}) : (0, jsx_runtime_1.jsx)(ChevronLeft_1.default, {}) }), isOpen && ((0, jsx_runtime_1.jsx)(material_1.Typography, { variant: "h6", gutterBottom: true, sx: { marginLeft: 1, marginBottom: 0 }, children: "Outline" }))] }), isOpen && (loading ? ((0, jsx_runtime_1.jsx)(material_1.Box, { sx: { display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100vh' }, children: (0, jsx_runtime_1.jsx)(material_1.CircularProgress, {}) })) : ((0, jsx_runtime_1.jsx)(material_1.List, { dense: true, children: outline.map((item, index) => ((0, jsx_runtime_1.jsx)(material_1.ListItem, { button: true, sx: { pl: item.level * 2 }, onClick: () => onItemClick(item.id), children: (0, jsx_runtime_1.jsx)(material_1.ListItemText, { primary: item.content, sx: {
                            overflow: 'hidden',
                            whiteSpace: 'nowrap',
                            textOverflow: 'ellipsis',
                        } }) }, item.id))) })))] }));
};
exports.default = Outline;
