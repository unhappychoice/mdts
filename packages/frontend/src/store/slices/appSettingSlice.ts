import { createSlice } from '@reduxjs/toolkit';

type DarkMode = 'dark' | 'light' | 'auto';
type ContentMode = 'full' | 'compact';

interface AppSettingState {
  darkMode: DarkMode;
  contentMode: ContentMode;
  fileTreeOpen: boolean;
  outlineOpen: boolean;
}

const initialState: AppSettingState = {
  darkMode: 'auto',
  contentMode: 'compact',
  fileTreeOpen: true,
  outlineOpen: true,
};

const appSettingSlice = createSlice({
  name: 'appSetting',
  initialState,
  reducers: {
    saveAppSetting: (state, action: PayloadAction<{ darkMode: DarkMode, contentMode: ContentMode }>) => {
      state.darkMode = action.payload.darkMode;
      state.contentMode = action.payload.contentMode;
    },
    toggleFileTree: (state) => {
      state.fileTreeOpen = !state.fileTreeOpen;
    },
    toggleOutline: (state) => {
      state.outlineOpen = !state.outlineOpen;
    },
  },
});

export const { saveAppSetting, toggleFileTree, toggleOutline } = appSettingSlice.actions;

export default appSettingSlice.reducer;
