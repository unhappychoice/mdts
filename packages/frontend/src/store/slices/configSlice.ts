import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
import { fetchData } from '../../api';

export const fetchConfig = createAsyncThunk('config/fetchConfig', async () => {
  return await fetchData<ConfigState>('/api/config', 'json');
});

export const saveConfigToBackend = createAsyncThunk(
  'config/saveConfigToBackend',
  async (config: ConfigState, { dispatch }) => {
    try {
      await fetch('/api/config', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(config),
      });
      dispatch(fetchConfig()); // Re-fetch config after saving
    } catch (error) {
      console.error('Failed to save settings:', error);
      throw error; // Re-throw to allow error handling in component
    }
  }
);

export interface ConfigState {
  fontFamily: string;
  fontFamilyMonospace: string;
  fontSize: number;
  syntaxHighlighterTheme: string;
}

const initialState: ConfigState = {
  fontFamily: 'Roboto',
  fontFamilyMonospace: 'monospace',
  fontSize: 14,
  syntaxHighlighterTheme: 'auto',
};

const configSlice = createSlice({
  name: 'config',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(fetchConfig.fulfilled, (state, action: PayloadAction<ConfigState>) => {
      state.fontFamily = action.payload.fontFamily;
      state.fontFamilyMonospace = action.payload.fontFamilyMonospace;
      state.fontSize = action.payload.fontSize;
      state.syntaxHighlighterTheme = action.payload.syntaxHighlighterTheme;
    });
  },
});

export default configSlice.reducer;
