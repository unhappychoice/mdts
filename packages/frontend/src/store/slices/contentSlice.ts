
import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
import { fetchData } from '../../api';

interface ContentState {
  content: string;
  loading: boolean;
  error: string | null;
  scrollPosition: number;
}

const initialState: ContentState = {
  content: '',
  loading: true,
  error: null,
  scrollPosition: 0,
};

export const fetchContent = createAsyncThunk(
  'content/fetchContent',
  async (path: string | null) => {
    const url = path ? `/api/markdown/${path}` : '/api/markdown/mdts-welcome-markdown.md';
    const data = await fetchData<string>(url, 'text');
    return data || '';
  }
);

const contentSlice = createSlice({
  name: 'content',
  initialState,
  reducers: {
    setScrollPosition: (state, action: PayloadAction<number>) => {
      state.scrollPosition = action.payload;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchContent.pending, (state) => {
        if (!state.content) {
          state.loading = true;
        }
        state.error = null;
      })
      .addCase(fetchContent.fulfilled, (state, action) => {
        state.loading = false;
        state.content = action.payload;
      })
      .addCase(fetchContent.rejected, (state, action) => {
        state.loading = false;
        state.error = action.error.message || 'Failed to fetch content';
      });
  },
});

export const { setScrollPosition } = contentSlice.actions;

export default contentSlice.reducer;
