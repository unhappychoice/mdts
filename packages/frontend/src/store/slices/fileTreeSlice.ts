
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { fetchData } from '../../api';

interface FileTreeItem {
  [key: string]: (FileTreeItem | string)[];
}

interface FileTreeState {
  fileTree: (FileTreeItem | string)[];
  filteredFileTree: (FileTreeItem | string)[];
  searchQuery: string;
  expandedNodes: string[];
  mountedDirectoryPath: string;
  loading: boolean;
  error: string | null;
}

const initialState: FileTreeState = {
  fileTree: [],
  filteredFileTree: [],
  searchQuery: '',
  expandedNodes: [],
  mountedDirectoryPath: '',
  loading: true,
  error: null,
};

const LOCAL_STORAGE_KEY_PREFIX = 'mdts_expanded_nodes_';
const LOCAL_STORAGE_RECENT_PATHS_KEY = 'mdts_recent_paths';
const MAX_RECENT_PATHS = 10;

const saveExpandedNodes = (path: string, nodes: string[]) => {
  try {
    localStorage.setItem(`${LOCAL_STORAGE_KEY_PREFIX}${path}`, JSON.stringify(nodes));
    let recentPaths: string[] = JSON.parse(localStorage.getItem(LOCAL_STORAGE_RECENT_PATHS_KEY) || '[]');
    recentPaths = recentPaths.filter(p => p !== path);
    recentPaths.unshift(path);
    if (recentPaths.length > MAX_RECENT_PATHS) {
      const oldPath = recentPaths.pop();
      if (oldPath) {
        localStorage.removeItem(`${LOCAL_STORAGE_KEY_PREFIX}${oldPath}`);
      }
    }
    localStorage.setItem(LOCAL_STORAGE_RECENT_PATHS_KEY, JSON.stringify(recentPaths));
  } catch (e) {
    console.error('Failed to save expanded nodes to local storage', e);
  }
};

const loadExpandedNodes = (path: string): string[] => {
  try {
    const storedNodes = localStorage.getItem(`${LOCAL_STORAGE_KEY_PREFIX}${path}`);
    return storedNodes ? JSON.parse(storedNodes) : [];
  } catch (e) {
    console.error('Failed to load expanded nodes from local storage', e);
    return [];
  }
};

const filterTree = (tree: (FileTreeItem | string)[], searchQuery: string): (FileTreeItem | string)[] => {
  if (!searchQuery) return tree;

  return tree
    .reduce((acc: (FileTreeItem | string)[], item) => {
      if (typeof item === 'string') {
        const fileName = item.split('/').pop() || '';
        return fileName.toLowerCase().includes(searchQuery.toLowerCase())
          ? [...acc, item]
          : acc;
      } else {
        const key = Object.keys(item)[0];
        const value = item[key];

        const children = Array.isArray(value)
          ? filterTree(value, searchQuery)
          : [];

        return children.length > 0
          ? [...acc, { [key]: children } as FileTreeItem ]
          :acc;
      }
    }, []);
};

export const fetchFileTree = createAsyncThunk(
  'fileTree/fetchFileTree',
  async () => {
    const data = await fetchData<{ fileTree: FileTreeItem[], mountedDirectoryPath: string }>('/api/filetree', 'json');
    return { fileTree: data?.fileTree || [], mountedDirectoryPath: data?.mountedDirectoryPath };
  }
);

const fileTreeSlice = createSlice({
  name: 'fileTree',
  initialState,
  reducers: {
    setSearchQuery: (state, action: { payload: string }) => {
      state.searchQuery = action.payload;
      state.filteredFileTree = filterTree(state.fileTree, action.payload);
    },
    toggleNode: (state, action: { payload: string }) => {
      const path = action.payload;
      if (state.expandedNodes.includes(path)) {
        state.expandedNodes = state.expandedNodes.filter((nodePath) => nodePath !== path);
      } else {
        state.expandedNodes.push(path);
      }
      saveExpandedNodes(state.mountedDirectoryPath, state.expandedNodes);
    },
    setExpandedNodes: (state, action: { payload: string[] }) => {
      state.expandedNodes = action.payload;
      saveExpandedNodes(state.mountedDirectoryPath, state.expandedNodes);
    },
    expandAllNodes: (state, action: { payload: (FileTreeItem | string)[] | null }) => {
      const allItemIds: string[] = [];
      const collectIds = (items: (FileTreeItem | string)[], parentPath: string = '') => {
        items.forEach(item => {
          if (typeof item !== 'string') {
            const key = Object.keys(item)[0];
            const currentPath = parentPath ? `${parentPath}/${key}` : key;
            allItemIds.push(currentPath);
            collectIds(item[key] as (FileTreeItem | string)[], currentPath);
          }
        });
      };
      if (action.payload) {
        collectIds(action.payload);
      }
      state.expandedNodes = allItemIds;
      saveExpandedNodes(state.mountedDirectoryPath, state.expandedNodes);
    },
    setMountedDirectoryPath: (state, action: { payload: string }) => {
      state.mountedDirectoryPath = action.payload;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchFileTree.pending, (state) => {
        state.loading = true;
        state.error = null;
      })
      .addCase(fetchFileTree.fulfilled, (state, action) => {
        state.loading = false;
        state.fileTree = action.payload.fileTree;
        state.mountedDirectoryPath = action.payload.mountedDirectoryPath;
        state.filteredFileTree = filterTree(action.payload.fileTree, state.searchQuery);
        state.expandedNodes = loadExpandedNodes(action.payload.mountedDirectoryPath);
      })
      .addCase(fetchFileTree.rejected, (state, action) => {
        state.loading = false;
        state.error = action.error.message || 'Failed to fetch file tree';
      });
  },
});

export const selectFilteredFileTree = (
  fullFileTree: (FileTreeItem | string)[],
  targetPath: string
): (FileTreeItem | string)[] => {
  if (targetPath === '') {
    const result: FileTreeItem[] | string[] = [];
    fullFileTree.forEach(item => {
      if (typeof item === 'string') {
        result.push(item.split('/').pop() || item);
      } else {
        const key = Object.keys(item)[0];
        const newObject: FileTreeItem = {};
        newObject[key.split('/').pop() || key] = item[key];
        result.push(newObject);
      }
    });
    return result;
  }

  const findChildren = (
    currentTree: (FileTreeItem | string)[],
    pathSegments: string[],
    currentSegmentIndex: number
  ): (FileTreeItem | string)[] | null => {
    if (currentSegmentIndex === pathSegments.length) {
      return currentTree;
    }

    const segment = pathSegments[currentSegmentIndex];

    for (const item of currentTree) {
      if (typeof item !== 'string') {
        const key = Object.keys(item)[0];
        const itemSegments = key.split('/');
        if (itemSegments[itemSegments.length - 1] === segment) {
          const children = item[key];
          if (Array.isArray(children)) {
            return findChildren(children, pathSegments, currentSegmentIndex + 1);
          }
        }
      }
    }
    return null;
  };

  const pathSegments = targetPath.split('/').filter(s => s !== '');
  const children = findChildren(fullFileTree, pathSegments, 0);

  if (children) {
    const result: FileTreeItem[] | string[] = [];
    children.forEach(item => {
      if (typeof item === 'string') {
        result.push(item.split('/').pop() || item);
      } else {
        const key = Object.keys(item)[0];
        const newObject: FileTreeItem = {};
        newObject[key.split('/').pop() || key] = item[key];
        result.push(newObject);
      }
    });
    return result;
  }

  return [];
};

export const {
  setSearchQuery,
  toggleNode,
  setExpandedNodes,
  expandAllNodes,
  setMountedDirectoryPath,
} = fileTreeSlice.actions;

export default fileTreeSlice.reducer;
