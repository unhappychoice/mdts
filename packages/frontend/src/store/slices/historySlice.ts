import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppDispatch } from '../store';

interface HistoryState {
  currentPath: string | null;
  isDirectory: boolean;
}

const initialState: HistoryState = {
  currentPath: null,
  isDirectory: false,
};

const historySlice = createSlice({
  name: 'history',
  initialState,
  reducers: {
    setHistory: (
      state,
      action: PayloadAction<{ path: string | null; isDirectory: boolean }>
    ) => {
      state.currentPath = action.payload.path;
      state.isDirectory = action.payload.isDirectory;
    },
  },
});

export const { setHistory } = historySlice.actions;

export const updateHistoryFromLocation = (pathname: string) => (dispatch: AppDispatch): void => {
  const getPathFromUrl = () => {
    const path = pathname.substring(1);
    if (path === '') return { path: null, isDirectory: false };

    const fileExtensions = ['.md', '.markdown'];
    const isFile = fileExtensions.some((ext) =>
      path.toLowerCase().endsWith(ext)
    );

    return { path: decodeURIComponent(path), isDirectory: !isFile };
  };

  const { path, isDirectory } = getPathFromUrl();
  dispatch(setHistory({ path, isDirectory }));
};

export default historySlice.reducer;
