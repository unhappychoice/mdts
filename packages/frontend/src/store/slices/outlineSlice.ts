import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { fetchData } from '../../api';

export type OutlineItem = {
  id: string;
  content: string;
  level: number;
};

interface OutlineState {
  outline: OutlineItem[];
  loading: boolean;
  error: string | null;
}

const initialState: OutlineState = {
  outline: [],
  loading: true,
  error: null,
};

export const fetchOutline = createAsyncThunk(
  'outline/fetchOutline',
  async (path: string | null) => {
    const url = path
      ? `/api/outline?filePath=${encodeURIComponent(path)}`
      : '/api/outline?filePath=mdts-welcome-markdown.md';
    const data = await fetchData<OutlineItem[]>(url, 'json');
    return data || [];
  }
);

const outlineSlice = createSlice({
  name: 'outline',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchOutline.pending, (state) => {
        state.loading = true;
        state.error = null;
      })
      .addCase(fetchOutline.fulfilled, (state, action) => {
        state.loading = false;
        state.outline = action.payload;
      })
      .addCase(fetchOutline.rejected, (state, action) => {
        state.loading = false;
        state.error = action.error.message || 'Failed to fetch outline';
      });
  },
});

export default outlineSlice.reducer;
