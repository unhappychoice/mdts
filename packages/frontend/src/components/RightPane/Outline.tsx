import { Box } from '@mui/material';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../../store/store';
import { fetchOutline } from '../../store/slices/outlineSlice';
import OutlineHeader from './OutlineHeader';
import OutlineContent from './OutlineContent';

interface OutlineProps {
  filePath: string;
  onItemClick: (id: string) => void;
  isOpen: boolean;
  onToggle: () => void;
}

const Outline: React.FC<OutlineProps> = ({ filePath, onItemClick, isOpen, onToggle }) => {
  const dispatch = useDispatch<AppDispatch>();
  const { outline, loading, error } = useSelector((state: RootState) => state.outline);

  useEffect(() => {
    dispatch(fetchOutline(filePath));
  }, [dispatch, filePath]);

  return (
    <Box sx={{
      width: isOpen ? '300px' : '66px',
      py: 2,
      bgcolor: 'background.paper',
      borderLeft: '1px solid',
      borderColor: 'divider',
      minHeight: '100%',
      flexShrink: 0,
    }}>
      <OutlineHeader isOpen={isOpen} onToggle={onToggle} />
      {isOpen && (
        <OutlineContent
          outline={outline}
          loading={loading}
          error={error}
          onItemClick={onItemClick}
        />
      )}
    </Box>
  );
};

export default Outline;
