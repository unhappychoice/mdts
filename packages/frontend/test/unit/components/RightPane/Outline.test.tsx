import { act, render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { thunk } from 'redux-thunk';
import Outline from '../../../../src/components/RightPane/Outline';
import { fetchOutline } from '../../../../src/store/slices/outlineSlice';

const mockStore = configureStore([thunk]);

jest.mock('../../../../src/store/slices/outlineSlice', () => ({
  ...jest.requireActual('../../../../src/store/slices/outlineSlice'),
  fetchOutline: jest.fn(() => ({ type: 'outline/fetchOutline/pending' })),
}));

describe('Outline', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      outline: {
        outline: [{ id: 'title', content: 'Title', level: 1 }],
        loading: false,
        error: null,
      },
    });
    store.dispatch = jest.fn();
  });

  test('renders correctly', async () => {
    let asFragment;
    await act(async () => {
      const { asFragment: f } = render(
        <Provider store={store}>
          <Outline filePath="/test.md" onItemClick={() => {}} isOpen={true} onToggle={() => {}} />
        </Provider>
      );
      asFragment = f;
    });
    expect(asFragment()).toMatchSnapshot();
  });

  test('dispatches fetchOutline on mount with correct filePath', () => {
    render(
      <Provider store={store}>
        <Outline filePath="/test.md" onItemClick={() => {}} isOpen={true} onToggle={() => {}} />
      </Provider>
    );
    expect(store.dispatch).toHaveBeenCalledWith(fetchOutline('/test.md'));
  });

  test('dispatches fetchOutline with null when filePath is null', () => {
    render(
      <Provider store={store}>
        <Outline filePath={null} onItemClick={() => {}} isOpen={true} onToggle={() => {}} />
      </Provider>
    );
    expect(store.dispatch).toHaveBeenCalledWith(fetchOutline(null));
  });

  test('calls onItemClick when an outline item is clicked', () => {
    const onItemClickMock = jest.fn();
    render(
      <Provider store={store}>
        <Outline filePath="/test.md" onItemClick={onItemClickMock} isOpen={true} onToggle={() => {}} />
      </Provider>
    );

    fireEvent.click(screen.getByText('Title'));
    expect(onItemClickMock).toHaveBeenCalledWith('title');
  });

  test('displays loading spinner when outline is loading', () => {
    store = mockStore({
      outline: {
        outline: [],
        loading: true,
        error: null,
      },
    });

    render(
      <Provider store={store}>
        <Outline filePath="/test.md" onItemClick={() => {}} isOpen={true} onToggle={() => {}} />
      </Provider>
    );

    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  test('displays error message when there is an error', () => {
    store = mockStore({
      outline: {
        outline: [],
        loading: false,
        error: 'Failed to load outline',
      },
    });

    render(
      <Provider store={store}>
        <Outline filePath="/test.md" onItemClick={() => {}} isOpen={true} onToggle={() => {}} />
      </Provider>
    );

    expect(screen.getByText('Error: Failed to load outline')).toBeInTheDocument();
  });
});
