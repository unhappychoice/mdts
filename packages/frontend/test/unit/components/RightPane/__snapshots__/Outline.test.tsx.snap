// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Outline renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-153qrh4"
  >
    <div
      class="MuiBox-root css-mzwsmu"
    >
      <button
        aria-label="toggle outline"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1nyqon6-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-onjnbs-MuiTypography-root"
      >
        Outline
      </h6>
    </div>
    <ul
      class="MuiList-root MuiList-padding MuiList-dense custom-scrollbar css-18l7s16-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-1s7t0kb-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense css-6qv7tz-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-2wbpcv-MuiTypography-root"
          >
            Title
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
